maxminddb/__init__.py

```diff
@@ -20,6 +20,19 @@
 from maxminddb.decoder import InvalidDatabaseError
 from maxminddb.reader import Reader as PyReader
 
+__all__ = [
+    "InvalidDatabaseError",
+    "MODE_AUTO",
+    "MODE_FD",
+    "MODE_FILE",
+    "MODE_MEMORY",
+    "MODE_MMAP",
+    "MODE_MMAP_EXT",
+    "PyReader",  # Exposed for type checking b/c return type of open_database()
+    "Reader",
```

Comment:
Rather than including these, perhaps we could export a particular type, e.g.:

```
ReaderTypes = Union[PyReader, "maxminddb.extension.Reader"]
```

I am not sure what the best naming would be here.

Reply:
i think it's best to expose a real base type. generally speaking, union return types are bad.

i suggest merging this as-is to simply establish the status quo, and cleaning up some things in another pull request.

for instance, maybe the oddly named Reader compat helper may be ditched in favour of something that can act as a proper type, and it may even be possible to do so without breaking API. i'll think about it some more

Reply:
see !88

Reply:
see #88